src/components/TopBar/ProjectPicker.jsx

```diff
@@ -0,0 +1,38 @@
+import map from 'lodash/map';
+import partial from 'lodash/partial';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ProjectPreview from '../../containers/ProjectPreview';
+import ProjectPickerButton from './ProjectPickerButton';
+import createMenu, {MenuItem} from './createMenu';
+
+const ProjectPicker = createMenu({
+  name: 'projectPicker',
+
+  isVisible({currentProjectKey, projectKeys}) {
+    return currentProjectKey && projectKeys.length > 1;
+  },
+
+  renderItems({currentProjectKey, projectKeys, onChangeCurrentProject}) {
+    return map(projectKeys, projectKey => (
+      <MenuItem
+        isEnabled={projectKey === currentProjectKey}
+        key={projectKey}
+        onClick={partial(onChangeCurrentProject, projectKey)}
+      >
+        <ProjectPreview projectKey={projectKey} />
+      </MenuItem>
+    ));
+  },
+})(ProjectPickerButton, ProjectPreview);
```

Comment:
Hmm does the second argument get used? It looks like these args go to `createMenuWithMappedProps`, which seems to have an arity of 1.

Reply:
OOPS YOU ARE RIGHT

Reply:
(My initial implementation took a second component argument which was used to render items, but that proved insufficiently flexible)